### ISSUE TITLE
How to calculate the TPR-95 accuracy ?

### ISSUE THREAD

**Issue by MTandHJ** [P2333/Rectified-Rejection#1]
Hi,
I read your paper about Rectified Confidence recently.
It's very interesting and compelling, especially the notion of T-Con oracle and the novel mild condition.
In utils.py, I found the implementation of calculation of TPR-95 accuracy, i.e., the 'calculate_FPR_TPR' function.
I feel confused about the last three lines:
```
    index_c = (torch.where(correct > T)[0]).size(0)
    index_w = (torch.where(wrong > T)[0]).size(0)
    acc = index_c / (index_c + index_w)
```
It seems that, TPR-95 accuracy is TP / (TP + FP). 
Could I assume that low-confidence decisions,  correct or wrong, are all excluded in this case?

In addition, before I read the codes, I thought a fixed threshold 1/2 would be applied to R-Con according to Theorem 1.
But the threshold $t$ in paper is exactly the 'T' in line 49 ?

Thanks.

**Comment by P2333** [comment]
Hi, thank you for your interestest.

When we calculate TPR-95 accuracy, 5% low-confidence or low-Rcon correct decisions are excluded, while some (either >5% or <5%) wrong decisions are also excluded.

In Theorem 1, a fixed threshold of 1/2 can make sure that the **filtered** correct/wrong samples are perfectly separated, but we cannot explicitly control the TPR rate. For example, there are about 2,700 to 3,700 correct samples kept after filtering in Figure 3, where the total number of correct samples is 5,200. Thus with a fixed threshold of 1/2 and different temperature \tau, the TPR rate is between 2,700/5,200 to 3,700/5,200.

In order to empirically compare with previous baselines, we control the TPR rate to be 95% and do not apply the confidence filter in the Experiments Section. We can treat Theorem 1/Figure 3 as a certified setting but cannot directly control the TPR rate; while the results in, e.g., Table 2 is an empirical setting that we can control the TPR rate but cannot obtain certified separability.

The threshold $t$ in paper is the 'T' in line 49, which could be 1/2 or other values as discussed above.

**Comment by MTandHJ** [comment]
Thanks for your detailed reply. 
Again, it's really a novel idea.
